site/cypress/e2e/cypress-feature/late_days_cache.spec.js

```diff
@@ -0,0 +1,349 @@
+import {buildUrl} from '/cypress/support/utils.js';
+
+const predictedStatus = (days_allowed, days_late, remaining) => {
+
+    if (days_late === 0) {
+        return 'Good';
+    }
+    else if (days_late <= days_allowed && days_late <= remaining) {
+        // Valid
+        return 'Late';
+    }
+    else if (days_late > remaining) {
+        // Bad (too many for term)
+        return 'Bad (too many late days used this term)';
+    }
+    else {
+        // Bad (too many for assignment)
+        return 'Bad (too many late days used on this assignment)';
+    }
+};
+
+const calculateCache = () => {
+    // Get cache recalculation request
+    cy.intercept('GET', buildUrl(['sample', 'bulk_late_days', 'calculate'])).as('calculateCache');
+
+    // Calculate all cache
+    cy.get('button').contains('Calculate Info').click();
+    cy.get('#rebuild-status-label').should('be.visible');
+
+    // Wait for query to finish
+    cy.wait('@calculateCache', {timeout: 300000});
+
+    // Wait for recalculation to finish
+    cy.get('#rebuild-status-label', {timeout: 15000}).should('not.be.visible');
+
+    for (const user_id of all_user_ids) {
+        cy.get(`[USER_ID="${user_id}"] > [ld_id="Late Days Remaining"]`)
+            .then((cell) => expect(cell.text().trim()).not.to.equal(''));
+    }
+};
+
+const checkStudentsInCache = () => {
+    cy.login('instructor');
```

Comment:
I don't think you need this here, since every call of this you are already logged in as instructor